pkg/cmd/auth/refresh/refresh.go

```diff
@@ -0,0 +1,127 @@
+package refresh
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/internal/config"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/pkg/prompt"
+	"github.com/spf13/cobra"
+)
+
+type RefreshOptions struct {
+	IO     *iostreams.IOStreams
+	Config func() (config.Config, error)
+
+	Hostname string
+	Scopes   []string
+}
+
+func NewCmdRefresh(f *cmdutil.Factory, runF func(*RefreshOptions) error) *cobra.Command {
+	opts := &RefreshOptions{
+		IO:     f.IOStreams,
+		Config: f.Config,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "refresh",
+		Args:  cobra.ExactArgs(0),
+		Short: "Request new scopes for a token",
+		Long: heredoc.Doc(`Expand the permission scopes for a given host's token.
+
+			This command allows you to add additional scopes to an existing authentication token via a web
+			browser. This enables gh to access more of the GitHub API, which may be required as gh adds
+			features or as you use the gh api command. 
+
+			Unfortunately at this time there is no way to add scopes without a web browser's involvement
+			due to how GitHub authentication works.
+
+			The --hostname flag allows you to operate on a GitHub host other than github.com.
+
+			The --scopes flag accepts a comma separated list of scopes you want to add to a token. If
+			absent, this command ensures that a host's token has the default set of scopes required by gh.
+
+			Note that if GITHUB_TOKEN is in the current environment, this command will not work.
+		`),
+		Example: heredoc.Doc(`
+			$ gh auth refresh --scopes write:org,read:public_key
+			# => open a browser to add write:org and read:public_key scopes for use with gh api
+
+			$ gh auth refresh
+			# => ensure that the required minimum scopes are enabled for a token and open a browser to add if not
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if runF != nil {
+				return runF(opts)
+			}
+
+			return refreshRun(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.Hostname, "hostname", "h", "", "The GitHub host to use for authentication")
```

Comment:
I wanted people to interactively select a host if none was passed and this seemed like the correct approach. if this default is specified, in `refreshRun` it would not be possible to tell if a user passed nothing (and thus I should prompt when multiple hosts are configured).

In an old style command I'd just check `.Changed()`, but we make a point of not accessing the flags directly in the `run` portion of commands.

Reply:
Makes sense! BTW, in cases when we still need to check `Changed()`, we do and save that information to an Options struct https://github.com/cli/cli/blob/26a5f118c16ffee53a8c08f9d0b09fb079099f5d/pkg/cmd/issue/create/create.go#L62-L63

But you have a good point about the interactive selection for a host happening, and in that case `github.com` is not the "default" (except that it likely appears first in the list).

Reply:
That was a valid path too (adding a *Provided) but I was seeing how not cluttering up the opts struct felt.